refinery/ui/source/js/file-browser/services/factory.spec.js

```diff
@@ -176,34 +176,72 @@
     describe('createColumnDefs', function () {
       var tempAssayAttributes = [
         { display_name: 'Attribute1',
-          internal_name: '_Attribute1_s'
+          internal_name: '_Attribute1_s',
+          attribute_type: 'Factor'
         },
         { display_name: 'Analysis Group',
-          internal_name: 'Analysis Group'
+          internal_name: 'Analysis Group',
+          attribute_type: 'Internal'
         },
-        { display_name: 'Url',
-          internal_name: 'Url'
+        { display_name: 'Download',
+          internal_name: 'Url',
+          attribute_type: 'Internal'
         }
       ];
+      var response = [];
+
+      beforeEach(inject(function () {
+        angular.copy(tempAssayAttributes, factory.assayAttributes);
+        response = factory.createColumnDefs();
+      }));
+
 
       it('createColumnDefs is a method', function () {
         expect(angular.isFunction(factory.createColumnDefs)).toBe(true);
       });
 
       it('returns an array of same length', function () {
-        angular.copy(tempAssayAttributes, factory.assayAttributes);
-        var response = factory.createColumnDefs();
         expect(response.length).toEqual(tempAssayAttributes.length);
       });
 
+      it('sets correct name', function () {
+        expect(response[0].name).toEqual(tempAssayAttributes[0].display_name);
+        expect(response[1].name).toEqual(tempAssayAttributes[1].display_name);
+      });
+
+      it('sets custom url name', function () {
+        expect(response[2].name).toEqual(tempAssayAttributes[2].internal_name);
+      });
```

Comment:
Should this be display_name too? It seems redundant, as-is.
```
0 + 1: display_name
2: internal_name
0 + 1: internal_name
2: internal_name
```

Reply:
No, the url is a custom method which grabs from the internal name and not the display name.